MergerService/src/run.cs

```diff
@@ -376,13 +386,25 @@ private void RunTask(MergeTask task, ITaskUtils taskUtils, string? managerCallba
                                     if (blob != null)
                                     {
                                         tiles.Add(new Tile(coord, blob));
+                                        currentBatchBytes += blob.Length;
+
+                                        // Flushes the "tiles" list if it reached the batch size or the batch max bytes
+                                        // This is done to prevent memory overflow
+                                        if (currentBatchBytes >= this._batchMaxBytes || (this._limitBatchSize && tiles.Count >= this._batchMaxSize))
+                                        {
```

Comment:
you forgot to update the  (tileProgressCount and overallTileProgressCount) if you eneterd the if and cleared tiles

Reply:
no need to clear them, tileProgressCount counts the tiles processed in the current batch
overallTileProgressCount counts the tiles processed in the current task
when we "flush" the processed tiles, we don't want to reset these variables